core/src/stores/postgres.rs

```diff
@@ -1782,24 +1804,37 @@ impl Store for PostgresStore {
             )
             .await?;
 
-        tx.query_one(
-            &stmt,
-            &[
-                &data_source_row_id,
-                &(document_created as i64),
-                &document_id,
-                &(document_timestamp as i64),
-                &document_tags,
-                &document_parents,
-                &document_source_url,
-                &document_hash,
-                &(document_text_size as i64),
-                &(document_chunk_count as i64),
-                &"latest",
-            ],
-        )
-        .await?;
+        let document_row_id = tx
+            .query_one(
+                &stmt,
+                &[
+                    &data_source_row_id,
+                    &(document_created as i64),
+                    &document_id,
+                    &(document_timestamp as i64),
+                    &document_tags,
+                    &document_parents,
+                    &document_source_url,
+                    &document_hash,
+                    &(document_text_size as i64),
+                    &(document_chunk_count as i64),
+                    &"latest",
+                ],
+            )
+            .await?
+            .get(0);
 
+        // TODO(KW_SEARCH_INFRA): make title/mime_type not optional.
+        // Upsert the data source node if title and mime_type are present. Otherwise, we skip the upsert.
+        if let (Some(_), Some(_)) = (document_title, document_mime_type) {
+            self.upsert_data_source_node(
```

Comment:
problem for now is the ID right ?

Reply:
seen IRL, sync between internalId and parents will be done in another task